packages/backend/src/lib/availability_cache.ts

```diff
@@ -0,0 +1,89 @@
+import type { DbClient } from "@esp-group-one/db-client";
+import { ObjectId } from "@esp-group-one/types";
+import type {
+  Availability,
+  AvailabilityCache,
+  Duration,
+} from "@esp-group-one/types";
+import type { Moment } from "moment";
+import moment from "moment";
+import type { Filter } from "mongodb";
+
+export async function getAvailabilityCache(
+  client: DbClient,
+  userId: ObjectId,
+): Promise<ObjectId[]> {
+  const coll = await client.availabilityCaches();
+  const res = await coll.find({ availablePeople: userId });
+
+  const availablePeople = res.flatMap((x) => x.availablePeople);
+  const uniqueAvailablePeople = Array.from(
+    new Set(availablePeople.map(String).filter((x) => x !== userId.toString())),
+  );
+
+  return uniqueAvailablePeople.map((x) => new ObjectId(x));
+}
+
+export async function setAvailabilityCache(
+  db: DbClient,
+  userId: ObjectId,
+  availability: Availability,
+): Promise<void> {
+  // Availablity is an objeect with start and end time. For each one hour time slot in the availabilty return me an time object. All the time objects should be an array
+  const start = moment(availability.timeStart).startOf("hour");
+  const end = moment(availability.timeEnd).startOf("hour");
+  //find the difference between start and end time
+  const duration = end.diff(start, "hours");
+  //create an array of time objects
```

Comment:
Please make sure the comments have spaces + capital letters
```suggestion
  // Create an array of time objects
```